torch/csrc/api/include/torch/optim/serialize.h

```diff
@@ -207,27 +207,77 @@ void serialize(
   }
 }
 
+template <typename T>
+c10::List<T> deque_to_list(const std::deque<T>& dq) {
+  c10::List<Tensor> list;
+  list.reserve(dq.size());
+  for (const auto& e : dq) {
+    list.push_back(e);
+  }
+  return list;
+}
+
+template <typename T>
+void list_to_deque(const c10::List<T>& list, std::deque<T>& dq) {
+  for (size_t i = 0; i < list.size(); i++) {
+    dq.push_back(list[i]);
+  }
+}
+
 #define _TORCH_OPTIM_SERIALIZE(name) \
   torch::optim::serialize(archive, #name, self.name)
 
 #define _TORCH_OPTIM_SERIALIZE_WITH_TEMPLATE_ARG(OptimizerName) \
   torch::optim::serialize<OptimizerName##ParamState, OptimizerName##Options>(archive, self)
 
-#define _TORCH_OPTIM_SERIALIZE_TORCH_ARG(name) \
-{ \
+#define _TORCH_OPTIM_SERIALIZE_TORCH_ARG(name) { \
   auto ivalue = torch::IValue(name()); \
   /* do not serialize if name is an undefined tensor*/ \
   if (!(ivalue.isTensor() && ivalue.unsafeToTensorImpl() == at::UndefinedTensorImpl::singleton())) { \
     archive.write(#name, ivalue); \
   } \
 }
 
+#define _TORCH_OPTIM_SERIALIZE_TORCH_ARG_DEQUE(name) {\
+  if (name().size() != 0) { \
```

Comment:
hmm is there a reason we don't serialize empty deque? (Is it a limitation from the IValue serialization side?)

Reply:
naah just changed it :D